src/overlays/actors/ovl_En_Horse_Ganon/z_en_horse_ganon.c

```diff
@@ -25,29 +45,291 @@ const ActorInit En_Horse_Ganon_InitVars = {
     (ActorFunc)EnHorseGanon_Update,
     (ActorFunc)EnHorseGanon_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Ganon/func_80A68660.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Ganon/func_80A686A8.s")
+extern SkeletonHeader D_06008668;
+extern AnimationHeader D_06004AA4;
+extern AnimationHeader D_06005264;
+extern AnimationHeader D_06005B78;
+extern AnimationHeader D_06002CE4;
+extern AnimationHeader D_06002650;
+extern AnimationHeader D_06003858;
+
+AnimationHeader* D_80A691B0[] = { &D_06004AA4, &D_06005264, &D_06005B78, &D_06002CE4 };
+AnimationHeader* D_80A691C0[] = { &D_06002650, &D_06003858 };
+static f32 animPlaybackSpeed[] = { 0.66666666f, 0.66666666f, 1.0f, 1.0f, 1.0f, 0.66666666f };
+static ColliderCylinderInit cylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x12, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 40, 100, 0, { 0, 0, 0 } },
+};
+static ColliderJntSphItemInit jntsphItemsInit[1] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 13, { { 0, 0, 0 }, 20 }, 100 },
+    },
+};
+static ColliderJntSphInit jntsphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x12, COLSHAPE_JNTSPH },
+    1,
+    jntsphItemsInit,
+};
+static CollisionCheckInfoInit collisionCheckInfoInit = {
+    0x0A,
+    0x0023,
+    0x0064,
+    0xFE,
+};
+s8 D_80A69248[] = { 0x09, 0xB8, 0x01, 0x26, 0x0E, 0x2C, 0x07, 0x00, 0x0C, 0x11, 0x01, 0x7A, 0x12, 0x69, 0x07, 0x00,
+                    0x06, 0x4E, 0xFE, 0xFB, 0x1D, 0xAC, 0x07, 0x00, 0x02, 0xF2, 0xFF, 0x45, 0x24, 0x4F, 0x07, 0x00,
+                    0xF9, 0x6E, 0xFE, 0x0C, 0x31, 0x22, 0x07, 0x00, 0xF3, 0x28, 0xFE, 0x0C, 0x32, 0xD5, 0x07, 0x00,
+                    0xEB, 0xEA, 0xFE, 0x5F, 0x2D, 0x6E, 0x07, 0x00, 0xE9, 0x5E, 0xFE, 0x27, 0x25, 0x65, 0x07, 0x00,
+                    0xE5, 0x93, 0xFE, 0x0C, 0x20, 0xAC, 0x07, 0x00, 0xE6, 0x25, 0xFE, 0x77, 0x1B, 0x07, 0x07, 0x00,
+                    0xEB, 0xB7, 0x00, 0x7C, 0x15, 0x39, 0x07, 0x00, 0xF4, 0x66, 0x00, 0x02, 0x11, 0xB9, 0x07, 0x00,
+                    0xF4, 0x7B, 0xFF, 0xDD, 0x11, 0xAF, 0x07, 0x00, 0xF8, 0x8D, 0xFF, 0xD1, 0x0B, 0xA2, 0x07, 0x00 };
+s32 D_80A692B8[] = { 0, 0x00000010 };
+static InitChainEntry initChain[] = {
+    ICHAIN_F32(unk_F8, 1200, ICHAIN_STOP),
+};
+void (*D_80A692C4[])(EnHorseGanon*, GlobalContext*) = { &func_80A68AF0, &func_80A68DB0, 0 };
+
+const f32 D_80A692D0 = 10430.3779f;
+
+void func_80A68660(Vec3s* data, s32 index, Vec3f* vec) {
+    Vec3s* temp = (Vec3s*)((u64*)data + index);
```

Comment:
feel like theres probably a way to clean this up without doing pointer math but I dont have any suggestions. maybe someone else knows

Reply:
I've tried to do this using array indexing notation but it changes the order of the parameters in the add instruction, so this really is the only way I've found to successfully do this.

Reply:
There is definitely something wrong with this, and with `D_80A69248` in general. To me it looks like it should either be a 2 dimensional array, or an array of some struct (possibly unknown so far).